connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -163,4 +164,29 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCTest
       assert(sql(s"SELECT char_length(c1) from $tableName").head().get(0) === 65536)
     }
   }
+
+  test("SPARK-46408: Scan with filter push-down with date time functions") {
```

Comment:
Because this is the first PR, I think it's OK to only push down `date_add` and `date_sub` for Mysql dialect.